dubbo-mock-extensions/dubbo-mock-admin/src/main/java/org/apache/dubbo/mock/filter/AdminMockFilter.java

```diff
@@ -0,0 +1,179 @@
+    
+    static {
+        ReferenceConfig<MockService> mockServiceConfig = new ReferenceConfig<>();
+        mockServiceConfig.setCheck(false);
+        mockServiceConfig.setInterface(MockService.class);
+        DubboBootstrap.getInstance().reference(mockServiceConfig);
+    }
+
+    public AdminMockFilter() {
+        typeHandler = new CommonTypeHandler();
+        Optional<DynamicConfiguration> dynamicConfigurationOptional = ApplicationModel.getEnvironment().getDynamicConfiguration();
+        if (!dynamicConfigurationOptional.isPresent()) {
+            logger.warn("[Dubbo Admin Mock] could not find configuration center, all consumer request will not be mocked!");
+            return;
+        }
+        DynamicConfiguration dynamicConfiguration = dynamicConfigurationOptional.get();
+        String config = dynamicConfiguration.getConfig(ADMIN_MOCK_RULE_KEY, ADMIN_MOCK_RULE_GROUP);
+        if (StringUtils.isNotEmpty(config)) {
+            GlobalMockRule newGlobalMockRule = new Gson().fromJson(config, GlobalMockRule.class);
+            globalMockRule.setEnableMock(newGlobalMockRule.getEnableMock());
+            globalMockRule.getEnabledMockRules().addAll(newGlobalMockRule.getEnabledMockRules());
+        }
+
+        dynamicConfiguration.addListener(ADMIN_MOCK_RULE_KEY, ADMIN_MOCK_RULE_GROUP, event -> {
+            globalMockRule.getEnabledMockRules().clear();
+            String content = event.getContent();
+            if (StringUtils.isBlank(content)) {
+                globalMockRule.setEnableMock(false);
+                return;
+            }
+            GlobalMockRule newRule = new Gson().fromJson(config, GlobalMockRule.class);
+            globalMockRule.setEnableMock(newRule.getEnableMock());
+            globalMockRule.getEnabledMockRules().addAll(newRule.getEnabledMockRules());
+        });
+    }
+
+    @Override
+    public Result invoke(Invoker<?> invoker, Invocation invocation) throws RpcException {
+        // check if open the admin mock config, global config.
+        if (!globalMockRule.getEnableMock()) {
+            return invoker.invoke(invocation);
+        }
+
+        String interfaceName = invocation.getTargetServiceUniqueName();
+        String methodName = invocation.getMethodName();
+        Object[] params = solveParams(invocation.getArguments());
+
+        // check if the service is in mock list
+        String mockRuleName = interfaceName + "#" + methodName;
+        Set<String> mockRules = globalMockRule.getEnabledMockRules();
+        if (CollectionUtils.isEmpty(mockRules) || !mockRules.contains(mockRuleName)) {
+            return invoker.invoke(invocation);
+        }
+        
+        // check if the MockService's invoker, then request.
+        if (Objects.equals(interfaceName, MockService.class.getName())) {
+            return invoker.invoke(invocation);
+        }
+        MockService mockService = DubboBootstrap.getInstance().getCache().get(MockService.class);
```

Comment:
Better use proxy factory to create mockService directly. Check how dubbo does that with MetadataService for example.